packages/cactus-test-cmd-api-server/src/test/typescript/integration/remote-plugin-imports.test.ts

```diff
@@ -27,86 +27,101 @@ import {
   PluginImportType,
 } from "@hyperledger/cactus-core-api";
 
-test("NodeJS API server + Rust plugin work together", async (t: Test) => {
-  const vaultTestContainer = new VaultTestServer({});
-  await vaultTestContainer.start();
+const testCase = "NodeJS API server + Rust plugin work together";
 
-  const ci = await Containers.getById(vaultTestContainer.containerId);
-  const vaultIpAddr = await Containers.getContainerInternalIp(ci);
-  t.comment(`Container VaultTestServer has IPv4: ${vaultIpAddr}`);
+describe(testCase, () => {
+  let vaultTestContainer: VaultTestServer;
+  let apiServer: ApiServer;
+  let pluginContainer: any;
 
-  test.onFinish(async () => {
-    await vaultTestContainer.stop();
-    await vaultTestContainer.destroy();
+  afterAll(async () => {
+    vaultTestContainer.stop();
+    vaultTestContainer.destroy();
+    apiServer.shutdown();
+    pluginContainer.stop();
+    pluginContainer.destroy();
   });
 
-  const hostPortVault = await vaultTestContainer.getHostPortHttp();
-  t.comment(`Container VaultTestServer (Port=${hostPortVault}) started OK`);
-  const vaultHost = `http://${vaultIpAddr}:${K_DEFAULT_VAULT_HTTP_PORT}`;
-
-  const pluginContainer = new CactusKeychainVaultServer({
-    envVars: [
-      `VAULT_HOST=${vaultHost}`,
-      `VAULT_TOKEN=${K_DEFAULT_VAULT_DEV_ROOT_TOKEN}`,
-      "HOST=0.0.0.0:8080",
-    ],
-  });
-  await pluginContainer.start();
-
-  test.onFinish(async () => {
-    await pluginContainer.stop();
-    await pluginContainer.destroy();
-  });
-
-  const hostPort = await pluginContainer.getHostPortHttp();
-  t.comment(`CactusKeychainVaultServer (Port=${hostPort}) started OK`);
-
-  const configuration = new Configuration({
-    basePath: `http://localhost:${hostPort}`,
-  });
-  const apiClient = new DefaultApi(configuration);
-
-  const configService = new ConfigService();
-  const apiServerOptions = await configService.newExampleConfig();
-  apiServerOptions.authorizationProtocol = AuthorizationProtocol.NONE;
-  apiServerOptions.configFile = "";
-  apiServerOptions.apiCorsDomainCsv = "*";
-  apiServerOptions.apiPort = 0;
-  apiServerOptions.cockpitPort = 0;
-  apiServerOptions.grpcPort = 0;
-  apiServerOptions.apiTlsEnabled = false;
-  apiServerOptions.plugins = [];
-  const config = await configService.newExampleConfigConvict(apiServerOptions);
-
-  const factory = new PluginFactoryKeychain({
-    pluginImportType: PluginImportType.Remote,
-  });
-
-  const plugin: IPluginKeychain = await factory.create({
-    keychainId: "_keychainId_",
-    instanceId: "_instanceId_",
-    remoteConfig: configuration,
+  test(testCase, async () => {
+    try {
```

Comment:
@Leeyoungone Why did the try/catch block needed to be added when in the original test code it wasn't there?

Reply:
I think jest got mad about not catching a failed promise or something (even though all the test were passing then too). So that's why I added the `try-catch` however, it works now (probably because I `awaited` everything within the afterAll ? (I'm not too sure but if it works I'm not gonna poke it).